pkg/operator/ceph/object/spec.go

```diff
@@ -381,8 +383,36 @@ func (c *clusterConfig) reconcileExternalEndpoint(cephObjectStore *cephv1.CephOb
 	return nil
 }
 
+func (c *clusterConfig) generateRoute(cephObjectStore *cephv1.CephObjectStore) (*routev1.Route, error) {
```

Comment:
We need to set the controller owner reference to the route as well, so it goes away when the CephObjectStore is gone.